controller/backing_image_controller.go

```diff
@@ -287,30 +287,34 @@ func (bic *BackingImageController) syncBackingImage(key string) (err error) {
 	return nil
 }
 
-func (bic *BackingImageController) IsBackingImageDataSourceCleanup(bi *longhorn.BackingImage) (finished bool, err error) {
+func (bic *BackingImageController) IsBackingImageDataSourceCleaned(bi *longhorn.BackingImage) (cleaned bool, requested bool, err error) {
 	bids, err := bic.ds.GetBackingImageDataSource(bi.Name)
-	if err != nil && !apierrors.IsNotFound(err) {
-		return false, err
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// can not get backing image data source, return cleaned so the controller won't wait until bids become failed-and-cleanup
+			return true, false, nil
+		}
+		return false, false, err
 	}
 
 	if bids.Spec.FileTransferred {
-		return true, nil
+		return true, false, nil
 	}
 
 	// when bids status is ready or ready for transfer, there is already no tmp file left
 	if bids.Status.CurrentState == longhorn.BackingImageStateFailedAndCleanUp ||
 		bids.Status.CurrentState == longhorn.BackingImageStateReady ||
 		bids.Status.CurrentState == longhorn.BackingImageStateReadyForTransfer {
-		return true, nil
+		return true, false, nil
 	}
 
-	// mark the status to failed so manager can clean up the orphan tmp file
-	bids.Status.Message = "backing image is deleted, try to cleaning up the tmp file of data source"
+	// mark the status to failed so manager can clean up the tmp file
+	bids.Status.Message = "backing image is deleted, requesting manager to clean up the tmp file of backing image data source"
 	bids.Status.CurrentState = longhorn.BackingImageStateFailed
 	if _, err = bic.ds.UpdateBackingImageDataSourceStatus(bids); err != nil {
-		return false, err
+		return false, false, err
 	}
-	return false, nil
+	return false, true, err
```

Comment:
`return false, true, nil` if we ensure there should not be any error if `requested` is true.